resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/config.scala

```diff
@@ -447,6 +447,13 @@ package object config extends Logging {
     .stringConf
     .createWithDefault("yarn.io/fpga")
 
+  private[spark] val APPLICATION_LIFETIME_TIMEOUT = ConfigBuilder("spark.yarn.app.lifetime.timeout")
+    .doc("Overall timeout for a spark application running on yarn from submitted time, " +
```

Comment:
Spark should be capitalized, YARN should be capitalized.

This should explicitly describe the LIFETIME timeout.  It doesn't look like there has been any work on adding other timeout values so I think hardcoding it to lifetime is sufficient for now.

Change to "from the time it was submitted.  By default it's unlimited"

Also add text about which versions of YARN support this option